packages/twenty-server/src/database/commands/command-runners/upgrade.command-runner.ts

```diff
@@ -0,0 +1,90 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import chalk from 'chalk';
+import { Repository } from 'typeorm';
+import { SemVer } from 'semver';
+import { isDefined } from 'twenty-shared';
+
+import {
+  ActiveOrSuspendedWorkspacesMigrationCommandRunner,
+  RunOnWorkspaceArgs,
+} from 'src/database/commands/command-runners/active-or-suspended-workspaces-migration.command-runner';
+import { EnvironmentService } from 'src/engine/core-modules/environment/environment.service';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { SyncWorkspaceMetadataCommand } from 'src/engine/workspace-manager/workspace-sync-metadata/commands/sync-workspace-metadata.command';
+import { isSameVersion } from 'src/utils/version/is-same-version';
+
+export abstract class UpgradeCommandRunner extends ActiveOrSuspendedWorkspacesMigrationCommandRunner {
+  abstract readonly fromVersion: SemVer;
+
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    protected readonly workspaceRepository: Repository<Workspace>,
+    protected readonly environmentService: EnvironmentService,
+    protected readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    protected readonly syncWorkspaceMetadataCommand: SyncWorkspaceMetadataCommand,
+  ) {
+    super(workspaceRepository, twentyORMGlobalManager, environmentService);
+  }
+
+  override async runOnWorkspace(args: RunOnWorkspaceArgs): Promise<void> {
+    const { workspaceId, index, total, options, appVersion } = args;
+
+    this.logger.log(
+      chalk.blue(
+        `${options.dryRun ? '(dry run)' : ''} Upgrading workspace ${workspaceId} ${index + 1}/${total}`,
+      ),
+    );
+
+    await this.validateWorkspaceVersionEqualFromVersionOrThrow({
```

Comment:
validateWorkspaceVersionEqualsVersionOrThrow

and we can exit this function in case workspaceVersion > fromVersion